### ISSUE TITLE
Initial value missing from django-rest-framework integration.

### ISSUE THREAD

**Issue by Psykar** [barseghyanartur/django-fobi#69]
Eg:
```javascript
"will_this_work": {
                "type": "boolean",
                "required": false,
                "read_only": false,
                "label": "Will this work?",
                "help_text": "y/n"
            },
```

Should be 
```javascript
"will_this_work": {
                "type": "boolean",
                "required": false,
                "read_only": false,
                "label": "Will this work?",
                "help_text": "y/n",
                "initial": false
            },
```

**Comment by Psykar** [comment]
Out of curiosity, why isn't the OPTIONS method just returning the same as the json export of a form?

**Comment by Psykar** [comment]
Scratch that... kind of.
Seems we get the metadata via `OPTIONS` but the initial data via `GET`
Fair I suppose, although vaguely annoying having to do two HTTP requests.

Exposing the json export of a form on the API somehow would be awesome still though.

**Comment by barseghyanartur** [comment]
@Psykar:

You're totally right. That's what I was thinking of within the last couple of days (as I'm busy studying React at the moment; there are plans to write a react-integration app). 

What you want is quite simple and is definitely gonna land in fobi soon (like, in a week of time) along with other useful information (such as max_value, min_value and even things like placeholders) - all of that will become available in OPTIONS.

**Comment by Psykar** [comment]
React integration is literally what I'm working on. Planning on using it
with https://github.com/mozilla-services/react-jsonschema-form
Just needs a relatively simple translation layer for the json.

**Comment by barseghyanartur** [comment]
@Psykar:

Fine. I was planning to use JSON schema for it, but didn't dive into it yet. Since you're doing that anyway, it would be great if you could write an overview of all meta-data that should land in the OPTIONS; per plugin.

**Presentational fields**

- content_image
- content_image_url
- content_text
- content_video

**Form fields**

- boolean
- checkbox_select_multiple
- date
- date_drop_down
- datetime
- decimal
- email
- file
- float
- hidden (in terms of the Django REST framework - a read-only field)
- input (some sort of a copy of ``text`` plugin)
- integer
- ip_address
- null_boolean
- password (some sort of a copy of ``text`` plugin)
- radio
- range_select
- regex
- select
- select_multiple
- select_multiple_with_max
- slider (just a copy of range_select, for compatibility with main package)
- slug
- text
- textarea (some sort of a copy of ``text`` plugin)
- time
- url

Could you do that?

**Comment by Psykar** [comment]
I'll take a look tomorrow, but I imagine so.

**Comment by barseghyanartur** [comment]
@Psykar:

Initial solution been implemented in [0.11.12](https://pypi.python.org/pypi/django-fobi/0.11.12).

**Some insights:**

Meta-data is passed in the ``DRFIntegrationFormElementPluginProcessor`` in ``field_metadata``, which is a dict.

- [Example 1: content_image plugin](https://github.com/barseghyanartur/django-fobi/blob/master/src/fobi/contrib/apps/drf_integration/form_elements/content/content_image/base.py#L54)

- [Example 2: decimal plugin](https://github.com/barseghyanartur/django-fobi/blob/master/src/fobi/contrib/apps/drf_integration/form_elements/fields/decimal/base.py#L86)

- [Example 3: decimal plugin](https://github.com/barseghyanartur/django-fobi/blob/master/src/fobi/contrib/apps/drf_integration/form_elements/fields/text/base.py#L55)

I have checked the JSON schema.

See the sample below.

```json
{
    "$schema": "http://json-schema.org/draft-04/schema#",
    "additionalProperties": true,
    "definitions": {},
    "id": "http://example.com/example.json",
    "properties": {
        "checked": {
            "default": false,
            "description": "An explanation about the purpose of this instance.",
            "id": "/properties/checked",
            "title": "The Checked Schema",
            "type": "boolean"
        },
        "id": {
            "default": 1,
            "description": "An explanation about the purpose of this instance.",
            "exclusiveMaximum": false,
            "exclusiveMinimum": false,
            "id": "/properties/id",
            "maximum": 300,
            "minimum": 50,
            "title": "The Id Schema",
            "type": "integer"
        },
        "name": {
            "default": "A green door",
            "description": "An explanation about the purpose of this instance.",
            "id": "/properties/name",
            "maxLength": 200,
            "minLength": 10,
            "title": "The Name Schema",
            "type": "string"
        },
        "price": {
            "default": 12.5,
            "description": "An explanation about the purpose of this instance.",
            "exclusiveMaximum": false,
            "exclusiveMinimum": false,
            "id": "/properties/price",
            "maximum": 300,
            "minimum": 50,
            "title": "The Price Schema",
            "type": "number"
        },
        "tags": {
            "additionalItems": true,
            "id": "/properties/tags",
            "items": {
                "default": "home",
                "description": "An explanation about the purpose of this instance.",
                "id": "/properties/tags/items",
                "maxLength": 200,
                "minLength": 10,
                "title": "The Empty Schema",
                "type": "string"
            },
            "type": "array",
            "uniqueItems": false
        }
    },
    "required": [
        "price",
        "tags",
        "checked",
        "id",
        "name"
    ],
    "type": "object"
}
```

I think it might make sense to keep things as they are, but I do want to support JSON schema as well. At the moment, how I see it, the best approach would be to make alternative ``FobiJSONSchemaMetaData`` class [here](https://github.com/barseghyanartur/django-fobi/blob/master/src/fobi/contrib/apps/drf_integration/metadata.py#L22). 

Would you be interested to work on that part? Pull requests are welcome.

**Comment by Psykar** [comment]
Started on this at https://github.com/Psykar/django-fobi/tree/feature/drf_json_schema

Unfortunately (for my work on this at least) the project I'm on has decided they're OK editing the json schema by hand. Personally this makes me a little happier and my work easier, but I likely won't have the time to complete this work, so if that WIP commit is useful as a starting point *shrug

**Comment by barseghyanartur** [comment]
@Psykar:

Yeah, that's definitely useful.

**Comment by barseghyanartur** [comment]
@Psykar:

I pulled your changes into [drf-json-schema](https://github.com/barseghyanartur/django-fobi/tree/drf-json-schema) branch, where I'll work on it further. 

Thanks!

**Comment by barseghyanartur** [comment]
@zackbleach:

I've just synced the drf-json-schema branch with master. Any work on this shall be done there. Your help is highly appreciated.
